libs/design-system/table/src/lib/table.component.ts

```diff
@@ -0,0 +1,74 @@
+import { CommonModule } from '@angular/common';
+import { AfterViewInit, Component, input, ViewChild } from '@angular/core';
+import { MatSort, MatSortModule, Sort } from '@angular/material/sort';
+import { MatTableDataSource, MatTableModule } from '@angular/material/table';
+
+/** Table data type */
+export interface TableData {
+  [name: string]: string | number;
```

Comment:
No need for this to be generic in a demo component. Add `name` and `value` properties